src/core/thread/network_data_tlvs.hpp

```diff
@@ -479,6 +491,42 @@ class PrefixTlv : public NetworkDataTlv
      */
     const uint8_t *GetPrefix(void) const { return reinterpret_cast<const uint8_t *>(this) + sizeof(*this); }
 
+    /**
+     * This method copies the Prefix from TLV into a given `Ip6::Prefix`.
+     *
+     * @param[out] aPrefix  An `Ip6::Prefix` to copy the Prefix from TLV into.
+     *
+     */
+    void CopyPrefixTo(Ip6::Prefix &aPrefix) const { aPrefix.Set(GetPrefix(), GetPrefixLength()); }
+
+    /**
+     * This method indicates whether the Prefix from TLV matches (same length and same prefix bytes) a given
+     * `Ip6::Prefix`.
+     *
+     * @param[in] aPrefix  A Prefix to match with.
+     *
+     * @retval TRUE   The TLV's Prefix matches @p aPrefix.
+     * @retval FALSE  The TLV's Prefix does not match @p aPrefix.
+     *
+     */
+    bool MatchesPrefix(Ip6::Prefix &aPrefix) const { return aPrefix.Matches(GetPrefix(), GetPrefixLength()); }
+
+    /**
+     * This method indicates whether the Prefix from TLV matches (same length and same prefix bytes) a given Prefix.
+     *
+     * @param[in]  aPrefix        A pointer to an IPv6 prefix.
+     * @param[in]  aPrefixLength  The prefix length pointed to by @p aPrefix (in bits).
+     *
+     * @retval TRUE   The TLV's Prefix matches @p aPrefix.
+     * @retval FALSE  The TLV's Prefix does not match @p aPrefix.
+     *
+     */
+    bool MatchesPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength) const
```

Comment:
Given length has to match, how about something like `IsEqual()`?
```suggestion
    bool IsEqual(const uint8_t *aPrefix, uint8_t aPrefixLength) const
```